cmd/habitat/community/consensus/cluster/cluster.go

```diff
@@ -0,0 +1,110 @@
+package cluster
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/url"
+	"os"
+
+	"github.com/eagraf/habitat/cmd/habitat/community/consensus/raft"
+	"github.com/eagraf/habitat/cmd/habitat/proxy"
+	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/rs/zerolog/log"
+)
+
+type ClusterService interface {
```

Comment:
just curious, why is this an interface? do you foresee making different implementations of this in the future?

Reply:
Yeah there is a good chance we implement other consensus implementations in the future. I don't think Raft will scale to hundreds of nodes ( we  might implement an alternative where only a handful of nodes in a community run raft ), or we might also do a Paxos or blockchain based version of consensus